sycl/test/extensions/annotated_usm/annotated_usm_negative.cpp

```diff
@@ -0,0 +1,87 @@
+  TEST((malloc_shared_annotated<int>), N, q, properties{conduit, foo{foo_enum::b}})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid runtime property in the property list.}}
+  TEST((malloc_host_annotated<int>), N, q, properties{conduit, foz{0, 1}})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid runtime property in the property list.}}
+  TEST((malloc_annotated<int>), N, q, alloc::device, properties{conduit, foo{foo_enum::a}, foz{0, 1}})
+}
+
+// Expect error when the property list contain invalid comile-time property
+// Note that two errors are raised for each case, during:
+// 1. validating malloc input properties
+// 2. validating annotated_ptr properties
+void testInvalidCompileTimeProperty(sycl::queue &q) {
+  // expected-error-re@sycl/ext/oneapi/experimental/common_annotated_properties/properties.hpp:* {{static assertion failed due to requirement {{.+}}: Property is invalid for the given type.}}
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid compile-time property in the property list.}}
+  TEST(malloc_device_annotated, N, q, properties{conduit, bar})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/common_annotated_properties/properties.hpp:* {{static assertion failed due to requirement {{.+}}: Property is invalid for the given type.}}
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid compile-time property in the property list.}}
+  TEST(malloc_shared_annotated, N, q, properties{conduit, baz<1>})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/common_annotated_properties/properties.hpp:* {{static assertion failed due to requirement {{.+}}: Property is invalid for the given type.}}
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid compile-time property in the property list.}}
+  TEST(malloc_host_annotated, N, q, properties{conduit, boo<double>})
+
+  // expected-error-re@sycl/ext/oneapi/experimental/common_annotated_properties/properties.hpp:* {{static assertion failed due to requirement {{.+}}: Property is invalid for the given type.}}
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Found invalid compile-time property in the property list.}}
+  TEST(malloc_annotated, N, q, alloc::device, properties{conduit, bar, baz<1>})
+}
+
+void testMissingUsmKind(sycl::queue &q) {
```

Comment:
Can you add test cases for the `aligned_alloc*` functions that require a usm_kind property?